cluster-operator/src/test/java/io/strimzi/operator/cluster/model/CruiseControlTest.java

```diff
@@ -0,0 +1,578 @@
+        String testEnvOneValue = "test.env.one";
+        envVar1.setName(testEnvOneKey);
+        envVar1.setValue(testEnvOneValue);
+
+        ContainerEnvVar envVar2 = new ContainerEnvVar();
+        String testEnvTwoKey = "TEST_ENV_2";
+        String testEnvTwoValue = "test.env.two";
+        envVar2.setName(testEnvTwoKey);
+        envVar2.setValue(testEnvTwoValue);
+
+        CruiseControlSpec cruiseControlSpec = new CruiseControlSpecBuilder()
+                .withImage(ccImage)
+                .withNewTemplate()
+                    .withNewCruiseControlContainer()
+                       .withEnv(envVar1, envVar2)
+                    .endCruiseControlContainer()
+                .endTemplate()
+                .build();
+
+        Kafka resource = ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image,
+                healthDelay, healthTimeout, metricsCm, kafkaConfig, zooConfig,
+                kafkaStorage, zkStorage, null, kafkaLogJson, zooLogJson, null, cruiseControlSpec);
+        CruiseControl cc = CruiseControl.fromCrd(resource, VERSIONS);
+
+        List<EnvVar> kafkaEnvVars = cc.getEnvVars();
+        assertThat(kafkaEnvVars.stream().filter(var -> testEnvOneKey.equals(var.getName())).map(EnvVar::getValue).findFirst().get(), is(testEnvOneValue));
+        assertThat(kafkaEnvVars.stream().filter(var -> testEnvTwoKey.equals(var.getName())).map(EnvVar::getValue).findFirst().get(), is(testEnvTwoValue));
+    }
+
+    @Test
+    public void testContainerTemplateEnvVarsWithKeyConflict() {
+        ContainerEnvVar envVar1 = new ContainerEnvVar();
+        String testEnvOneKey = "TEST_ENV_1";
+        String testEnvOneValue = "test.env.one";
+        envVar1.setName(testEnvOneKey);
+        envVar1.setValue(testEnvOneValue);
+
+        ContainerEnvVar envVar2 = new ContainerEnvVar();
+        String testEnvTwoKey = "TEST_ENV_2";
+        String testEnvTwoValue = "my-special-value";
+        envVar2.setName(testEnvTwoKey);
+        envVar2.setValue(testEnvTwoValue);
+
+        CruiseControlSpec cruiseControlSpec = new CruiseControlSpecBuilder()
+                .withImage(ccImage)
+                .withNewTemplate()
+                    .withNewCruiseControlContainer()
+                        .withEnv(envVar1, envVar2)
+                    .endCruiseControlContainer()
+                .endTemplate()
+                .build();
+
+        Kafka resource = ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image,
+                healthDelay, healthTimeout, metricsCm, kafkaConfig, zooConfig,
+                kafkaStorage, zkStorage, null, kafkaLogJson, zooLogJson, null, cruiseControlSpec);
+        CruiseControl cc = CruiseControl.fromCrd(resource, VERSIONS);
+
+        List<EnvVar> kafkaEnvVars = cc.getEnvVars();
+        assertThat(kafkaEnvVars.stream().filter(var -> testEnvOneKey.equals(var.getName())).map(EnvVar::getValue).findFirst().get(), is(testEnvOneValue));
```

Comment:
`assertThat(kafkaEnvVars, hasEntry(testEnvOneKey, testEnvOneValue)` suggested improvement
And maybe have same for envar2?

Reply:
Since `kafkaEnvVars` is a list instead of a map I used `hasItem()` instead. How does this look:

`
List<EnvVar> envVarList = cc.getEnvVars();

assertThat(envVarList, hasItems(new EnvVar(testEnvOneKey, testEnvOneValue, null)));
assertThat(envVarList, hasItems(new EnvVar(testEnvTwoKey, testEnvTwoValue, null)));
`
Let me know what you think!

Reply:
Looks good, though since hasItems takes `...items` you could achieve the same with 
```
assertThat(envVarList, hasItems(new EnvVar(testEnvOneKey, testEnvOneValue, null),
    new EnvVar(testEnvTwoKey, testEnvTwoValue, null)));
```

But this is only a suggestion , not a required change at all! :P